src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/DashboardItemOrder.tsx

```diff
@@ -168,17 +186,126 @@ export const DashboardItemOrder = ({
+  const intl = useIntl();
+  const { organization } = order;
+  if (!organization) {
+    return null;
+  }
+
+  const showContactsBlock =
+    organization.contact_email || organization.contact_phone || organization.dpo_email;
+
+  return (
+    <div className="dashboard-splitted-card mt-s">
+      <div className="dashboard-splitted-card__column">
+        <ContractItem order={order} product={product} />
+        {showContactsBlock && (
+          <div className="dashboard-splitted-card__item">
+            <div className="dashboard-splitted-card__item__title">Contacts</div>
+            <div className="dashboard-splitted-card__item__description">
+              {organization.contact_email && (
+                <div>
+                  Main
+                  <Button
+                    size="small"
+                    color="tertiary"
+                    href={'mailto:' + (organization.contact_email ?? '')}
+                  >
+                    {organization.contact_email}
+                  </Button>
+                </div>
+              )}
+              {organization.contact_phone && (
+                <div>
+                  Phone
+                  <Button
+                    size="small"
+                    color="tertiary"
+                    href={'tel:' + (organization.contact_phone ?? '')}
+                  >
+                    {organization.contact_phone}
+                  </Button>
+                </div>
+              )}
+              {organization.dpo_email && (
+                <div>
+                  DPO
+                  <Button
+                    size="small"
+                    color="tertiary"
+                    href={'mailto:' + (organization.dpo_email ?? '')}
+                  >
+                    {organization.dpo_email}
+                  </Button>
+                </div>
+              )}
+            </div>
+          </div>
+        )}
+        {organization.address && (
+          <div className="dashboard-splitted-card__item dashboard-splitted-card__item__address">
+            <div className="dashboard-splitted-card__item__title">Address</div>
+            <div className="dashboard-splitted-card__item__description">
+              <AddressView address={organization.address} />
+            </div>
+          </div>
+        )}
+      </div>
+      <div className="dashboard-splitted-card__column">
+        <div className="dashboard-item-order__organization">
+          <div className="dashboard-item-order__organization__header">
+            <FormattedMessage {...messages.organizationHeader} />
+          </div>
+          <img
+            src={organization.logo?.src}
+            srcSet={organization.logo?.srcset}
+            alt={intl.formatMessage(messages.organizationLogoAlt)}
+          />
+          <div className="dashboard-item-order__organization__name">{organization.title}</div>
+        </div>
+      </div>
```

Comment:
Agreed, as you will see I had to rewrite the structure of this block to introduce a `separator` otherwise the border separator of one of the columns does not work anymore due to reversed order and the usage of `:first-child`